pkg/web/oauthclient/cookie_state.go

```diff
@@ -35,13 +35,19 @@ func init() {
 }
 
 // StateCookie returns the cookie storing the state of the console.
-func (oc *OAuthClient) StateCookie() *cookie.Cookie {
+func (oc *OAuthClient) StateCookie(tls bool) *cookie.Cookie {
+	sameSite := http.SameSiteNoneMode
+	if !tls {
+		// SameSite=None will be rejected when not using TLS, so we have to
+		// use SameSite=Lax in this case.
+		sameSite = http.SameSiteLaxMode
+	}
```

Comment:
Again, I don't think this should be "magic", so I'd prefer to make this a `CrossSiteCookie bool` in `pkg/web/oauthclient.Config`. It can be `false` by default, and we can set it to `true` in multi-cluster deployments and for OIDC.